src/main/java/lotto/util/InputValidator.java

```diff
@@ -0,0 +1,108 @@
+package lotto.util;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import lotto.model.Lotto;
+import lotto.view.error.ErrorType;
+import lotto.view.error.InputErrorException;
+
+public class InputValidator implements Validator{
+
+    private static final String WHITE_SPACE_REGEX = "\\s";
+    private static final String EMPTY_STRING = "";
+    private static final int MIN_LOTTO_NUMBER = 1;
+    private static final int MAX_LOTTO_NUMBER = 45;
+    private static final int MIN_PURCHASE_AMOUNT = 1000;
+    private static final int MAX_PURCHASE_AMOUNT = 10000000;
+    private static final String NEED_NUMBER_IN_RANGE = "구매 금액은 1000이상 10000000이하여야 합니다.";
+
+
+    @Override
+    public void checkIfEmpty(String input){
+        if (input.isEmpty()) {
+            throw new InputErrorException(ErrorType.NEED_NOT_EMPTY);
+        }
+        if (input.replaceAll(WHITE_SPACE_REGEX, EMPTY_STRING).isEmpty()) {
+            throw new InputErrorException(ErrorType.NEED_NOT_EMPTY);
+        }
+    }
+
+    @Override
+    public void checkIfEmpty(List<Integer> input){
+        if (input.isEmpty()) {
+            throw new InputErrorException(ErrorType.NEED_NOT_EMPTY);
+        }
+    }
+
+
+    @Override
+    public void checkIfDigit(String input){
+        for (int i = 0; i < input.length(); i++) {
+            if (!Character.isDigit(input.charAt(i))) {
+                throw new InputErrorException(ErrorType.NEED_NUMBER);
+            }
+        }
+    }
+
+    @Override
+    public void validateOnlyDigit(List<String> lottoNumbers){
+        lottoNumbers.forEach(lottoNumber -> {
+            checkIfEmpty(lottoNumber);
+            checkIfDigit(lottoNumber);
+        });
+    }
+
+
+    @Override
+    public void validateOnlyDigit(String input){
+        checkIfEmpty(input);
+        checkIfDigit(input);
+    }
+
+    @Override
+    public void checkValidRange(Lotto lotto){
+        lotto.getLottoNumbers().forEach(lottoNumber -> {
+            if (lottoNumber < MIN_LOTTO_NUMBER || lottoNumber > MAX_LOTTO_NUMBER) {
+                throw new InputErrorException(ErrorType.NEED_NUMBER_IN_RANGE);
+            }
+        });
```

Comment:
바로 밑에 정의한 메서드를 재사용해보는건 어떨까요?
```suggestion
        lotto.getLottoNumbers().forEach(this::checkValidRange);
```

Reply:
그러네요!! 와 동일한 로직이니 반복할 필요없이 메서드를 가져와서 사용하면 될 것 같습니다. 감사합니다!